graphix/pauli.py

```diff
@@ -0,0 +1,285 @@
+        return result
+
+    def prefix(self, s: str) -> str:
+        """
+        Prefix the given string by the complex unit as coefficient,
+        1 leaving the string unchanged.
+        """
+        if self.__im:
+            result = "1j*" + s
+        else:
+            result = s
+        if self.__sign:
+            result = "-" + result
+        return result
+
+    def __mul__(self, other):
+        if isinstance(other, ComplexUnit):
+            im = self.__im != other.__im
+            sign = (self.__sign != other.__sign) != (self.__im and other.__im)
+            return COMPLEX_UNITS[sign][im]
+        return NotImplemented
+
+    def __rmul__(self, other):
+        if other == 1:
+            return self
+        elif other == -1:
+            return COMPLEX_UNITS[not self.__sign][self.__im]
+        elif other == 1j:
+            return COMPLEX_UNITS[self.__sign != self.__im][not self.__im]
+        elif other == -1j:
+            return COMPLEX_UNITS[self.__sign == self.__im][not self.__im]
+
+    def __neg__(self):
+        return COMPLEX_UNITS[not self.__sign][self.__im]
+
+
+COMPLEX_UNITS = [[ComplexUnit(sign, im) for im in (False, True)] for sign in (False, True)]
+
+
+UNIT = COMPLEX_UNITS[False][False]
+
+
+UNITS = [UNIT, -UNIT, 1j * UNIT, -1j * UNIT]
+
+
+class Axis(enum.Enum):
+    X = 0
+    Y = 1
+    Z = 2
+
+
+class Plane(enum.Enum):
+    XY = 0
+    YZ = 1
+    XZ = 2
+
+    @property
+    def axes(self) -> list[Axis]:
+        match self:
+            case Plane.XY:
+                return [Axis.X, Axis.Y]
+            case Plane.YZ:
+                return [Axis.Y, Axis.Z]
+            case Plane.XZ:
+                return [Axis.X, Axis.Z]
+
+    @property
+    def cos(self) -> Axis:
+        match self:
+            case Plane.XY:
+                return Axis.X
+            case Plane.YZ:
+                return Axis.Z  # former convention was Y
+            case Plane.XZ:
+                return Axis.Z  # former convention was X
+
+    @property
+    def sin(self) -> Axis:
+        match self:
+            case Plane.XY:
+                return Axis.Y
+            case Plane.YZ:
+                return Axis.Y  # former convention was Z
+            case Plane.XZ:
+                return Axis.X  # former convention was Z
+
+    def polar(self, angle: float) -> tuple[float, float, float]:
+        result = [0, 0, 0]
+        result[self.cos.value] = np.cos(angle)
+        result[self.sin.value] = np.sin(angle)
+        return tuple(result)
+
+    @staticmethod
+    def from_axes(a: Axis, b: Axis) -> "Plane":
+        if b.value < a.value:
+            a, b = b, a
+        match a, b:
+            case Axis.X, Axis.Y:
+                return Plane.XY
+            case Axis.Y, Axis.Z:
+                return Plane.YZ
+            case Axis.X, Axis.Z:
+                return Plane.XZ
+        assert a == b
+        raise ValueError(f"Cannot make a plane giving the same axis {a} twice.")
+
```

Comment:
comment from @mgarnier59 : this convention is consistent with Backens et al., 
to @masa10-f : I think this is much clearer than what we had before- any comments?

Reply:
It seems nice. No comment so far :)